symbolic-uplc/src/ZkFold/Symbolic/UPLC/Evaluation.hs

```diff
@@ -253,7 +299,28 @@ applyPoly ctx SndPair [arg] = do
   MaybeValue p <- evalArg ctx arg []
   (_, ExValue v) <- asPair (Symbolic.fromJust p)
   return $ MaybeValue (symMaybe p v)
-applyPoly _ (BPFList _) _ = error "FIXME: UPLC List support"
+applyPoly ctx (BPFList ChooseList) (ct:tt:et:args) = let
+    b0 = applyPoly ctx (BPFList NullList) [ct]
+    bt =   evalArg ctx (AThunk . Right $ b0) []
+  in applyPoly ctx IfThenElse ((AThunk . Right $ bt):tt:et:args)
+applyPoly ctx (BPFList MkCons) [x0,xs0] = do
+  MaybeValue x <- evalArg ctx x0 []
+  let l = Symbolic.fromJust x
+  MaybeValue xs <- evalArg ctx xs0 []
+  let ls = Symbolic.fromJust xs
+  return $ MaybeValue (symMaybe2 x xs $ l L..: (fromJust $ cast ls))
+applyPoly ctx (BPFList HeadList) (xs0:args) = do
+  MaybeValue p <- evalArg ctx xs0 args
```

Comment:
No, `args` should be applied to the head, not to the whole list
For now it would be better to just say that we don't want any more arguments with `[xs0]` in the pattern match then, I guess